homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Flux lights."""
-    config_auto = entry.options["global"].get(
-        CONF_AUTOMATIC_ADD, entry.data[CONF_AUTOMATIC_ADD]
-    )
-    config_devices = entry.data[CONF_DEVICES]
-    config_options = entry.options
-
-    lights = []
-
-    if config_auto:
-        # Find the bulbs on the LAN
-        scanner = BulbScanner()
-        await hass.async_add_executor_job(scanner.scan)
-
-        for device in scanner.getBulbInfo():
-            device_id = device["ipaddr"].replace(".", "_")
-            if device_id not in config_devices:
-                config_devices[device_id] = device
-
-    for device_id, device in config_devices.items():
-        add_device = {}
-        add_device["name"] = device.get("name", device_id)
-        add_device[CONF_HOST] = device[CONF_HOST]
-        add_device[CONF_PROTOCOL] = None
-        add_device[ATTR_MODE] = None
-        add_device[CONF_CUSTOM_EFFECT] = None
-        add_device[CONF_EFFECT_SPEED] = config_options.get(device_id, {}).get(
-            CONF_EFFECT_SPEED,
-            config_options.get("global", {}).get(
-                CONF_EFFECT_SPEED, DEFAULT_EFFECT_SPEED
-            ),
+
+    async def async_new_lights(bulbs: dict):
+        """Add new bulbs when they are found or configured."""
+        bulb_coordinator = (
```

Comment:
The coordinator doesn't need to be nested in this callback. It can be accessed from the outer scope of the callback as long as we create that variable in `async_setup_entry`.

Reply:
Not sure what you mean on this one. If I declare bulb_coordinator in the main `async_setup_entry` above this callback I don't need it here?

Reply:
Actually this gets removed completely so I think this is moot.